internal/controllers/machine/machine_controller_noderef.go

```diff
@@ -116,6 +118,16 @@ func (r *Reconciler) reconcileNode(ctx context.Context, cluster *clusterv1.Clust
 		}
 	}
 
+	options := []client.PatchOption{
+		client.FieldOwner("capi-machine"),
```

Comment:
```suggestion
		client.FieldOwner("capi-machine"),
		client.ForceOwnership,
```
When using SSA in controllers it is recommended to always force conflicts.